system/modules/core/library/Contao/Controller.php

```diff
@@ -1071,9 +1052,40 @@ public static function redirect($strLocation, $intStatus=303)
 			header('Location: ' . $strLocation);
 		}
 
+		static::setCorsHeadersForCrossDomainSearchIndexRebuild();
+
 		exit;
 	}
 
+	/**
+	 * Sends the correct HTTP headers for the back end rebuild search index feature.
+	 * To be removed as soon as we have a real web spider.
+	 */
+	public function setCorsHeadersForCrossDomainSearchIndexRebuild()
+	{
+		// OPTIONS preflight request
+		if (\Environment::get('requestMethod') == 'OPTIONS' && strpos(strtolower(\Environment::get('httpAccessControlRequestHeaders')), 'contao-rebuild-search-index') !== false)
+		{
+			header('Access-Control-Allow-Headers: X-Requested-With, Contao-Rebuild-Search-Index');
+			header('Access-Control-Allow-Methods: GET, OPTIONS');
+			header('Access-Control-Allow-Origin: ' . \Environment::get('httpOrigin'));
+		}
+
+		// GET request (protect to only allow requests from own known hosts)
+		if (\Environment::get('requestMethod') == 'GET' && $strOrigin = \Environment::get('httpContaoRebuildSearchIndex'))
+		{
+			$blnCheck = \Database::getInstance()->prepare('SELECT id FROM tl_page WHERE type=? AND dns=?')
+				->execute('root', preg_replace('@^https?://@', '', $strOrigin));
```

Comment:
It is required. Remember, this PR resolves **cross-domain** issues. You cannot have multiple domains in one installation without entering them in the root pages ;-)